src/utils/tracks.js

```diff
@@ -363,7 +363,81 @@ export const getTimeOfDayPeriodBasedOnTime = (datetimeString, timeZone) => {
   const [hour, min] = getTimeInTimezone(new Date(datetimeString), timeZone).split(':');
   const trackTotalMinutesInTZ = ( (parseInt(hour) * 60) + parseInt(min) ) || 1440;
 
-  return TIME_OF_DAY_PERIODS.findIndex((timeOfDayPeriod) =>
+  const period = TIME_OF_DAY_PERIODS.find((timeOfDayPeriod) =>
     trackTotalMinutesInTZ >= timeOfDayPeriod.rangeMinutesMin && trackTotalMinutesInTZ <= timeOfDayPeriod.rangeMinutesMax
   );
+
+  return period ?? TIME_OF_DAY_PERIODS[0];
+};
+
+export const buildFeatureCollectionOfTwoPointLineStringSegments = (trackFeatureCollection, timeZone) => {
+  const featureCollection = {
+    type: 'FeatureCollection',
+    features: []
+  };
+
+  if (!trackFeatureCollection || !trackFeatureCollection.features || !trackFeatureCollection.features.length) {
+    return featureCollection;
+  }
+
+  const [lineStringFeature] = trackFeatureCollection.features;
+
+  // Checking if first feature is valid
+  if (!lineStringFeature?.geometry ||
+    lineStringFeature.geometry?.type !== 'LineString' ||
+    !lineStringFeature.properties?.coordinateProperties?.times) {
+    return featureCollection;
+  }
+
+  const {
+    geometry: {
+      coordinates
+    },
+    properties: {
+      coordinateProperties: {
+        times
+      }
+    }
+  } = lineStringFeature;
+
+  // At least there should be 2 points to create the line string and the amount of times should be the same as the amount of coordinates
+  if (coordinates.length < 2 || coordinates.length !== times.length) {
+    return featureCollection;
+  }
+
+  const segments = [];
+
+  // Create a line segment for each pair of consecutive points
+  for (let i = 0; i < coordinates.length - 1; i++) {
+    const startTime = times[i];
+    const endTime = times[i + 1];
+    const startCoors = coordinates[i];
+    const endCoors = coordinates[i + 1];
+
+    if (!endTime || !endCoors){
+      break;
+    }
+
+    const { color: startColor } = getTimeOfDayPeriodBasedOnTime(startTime, timeZone);
+    const { color: endColor } = getTimeOfDayPeriodBasedOnTime(endTime, timeZone);
+
+    segments.push({
+      type: 'Feature',
+      properties: {
+        startColor,
+        endColor,
+        startTime,
+        endTime
+      },
+      geometry: {
+        type: 'LineString',
+        coordinates: [startCoors, endCoors]
+      }
+    });
+  }
+
+  return {
```

Comment:
This file already imports the `featureCollection` utility fn from turf, why not just `return featureCollection(segments)` ? Seems cleaner to me.